activerecord/lib/active_record/migration.rb

```diff
@@ -681,6 +693,18 @@ def migrate(direction)
       def disable_ddl_transaction!
         @disable_ddl_transaction = true
       end
+
+      private
+        def check_pending_migrations_unsafely
+          prev_db_config = Base.connection_db_config
+          all_configs = ActiveRecord::Base.configurations.configs_for(env_name: Rails.env)
```

Comment:
In this one we could pass `all_configs` to `check_pending_migrations` since we have them in `load_schema_if_pending!`.

The method `load_schema_if_pending!` is only called from parallel testing which is why it's safe to use `Rails.env` here. It has to load the env for that to work, but in the other location we need to follow the contract that Active Record works without Rails.